### ISSUE TITLE
LSL launch and connectivity with OpenVibe

### ISSUE THREAD

**Issue by hkn1304** [sccn/labstreaminglayer#183]
Hello,
I cannot see the exe files and launch LSL from Matlab. Is there a structured step by step tutorial on what to do from scrath to configuring in Matlab so that people who doesn't know much about coding world and visual studio like me can also utilize this in their research.
Please help!

**Comment by cboulay** [comment]
There are no exe files needed.

Did you see the Matlab examples folder? https://github.com/sccn/labstreaminglayer/tree/master/LSL/liblsl-Matlab/examples

**Comment by hkn1304** [comment]
Yes, i saw it. However the problem is when i start Lab Recorder, i cannot link any of Apps(mouse, keyboard etc) because i can not see exe files for example in mouse app folder allowing me to link. 
![image](https://cloud.githubusercontent.com/assets/22224352/24332514/77b5497a-1250-11e7-9d17-44ff05dd70d2.png)

I think because of the problem i described above, when i run ReceiveData.m in Matlab it just says "Resolving an EEG stream". 

![image](https://cloud.githubusercontent.com/assets/22224352/24332524/90422918-1250-11e7-9063-c5b6e26dddab.png)

I am stucked at this point and lost a week from now on. Please help.

**Comment by cboulay** [comment]
Right. For ReceiveData or LabRecorder to be able to resolve a stream, there must be a stream on the network. To put a stream on the network, you need an LSL App to poll a data source and push the data to the stream via the Outlet.

The Mouse application is one potential Outlet Application. It looks like you tried to build this already. Was the build unsuccessful? Look in the release or debug folders for the .exe. But you might not have to build an app. Some are already built for you.

If you follow the tutorials [here](https://github.com/sccn/labstreaminglayer/wiki/Tutorial), you'll see that a bunch of the links go to the ftp, where [pre-built apps](ftp://sccn.ucsd.edu/pub/software/LSL/Apps/) are available.

**Comment by dmedine** [comment]
Under the section 'Download Binary Releases' is a link to the binary 
releases (.exe).

ftp://sccn.ucsd.edu/pub/software/LSL/

**Comment by hkn1304** [comment]
Thanks gents. As you say i downloaded the exe versions which i am using properly now.

One more thing, is there a way to make script (apart from vis_stream) to get real time raw EEG data to  store in matrix form (.mat file) and plot? so that i can incorporate into my ANN based EEG code(written by myself?

Lastly, i would like to control let's say a toy car(arduino based) . if i use online BCILAB analysis, is there a way to easily designate processed Right/Left hand imagery to control the car? Any simulink connection?
Thanks very much.

**Comment by cboulay** [comment]
>real time raw EEG data to store in matrix form (.mat file) and plot

I'm not sure what you're asking here. Do you want access to the real-time data in Matlab? Then use the liblsl-Matlab library and create an inlet. Do you want to store the data to disc and load it later in Matlab? Then use LabRecorder to write to an xdf file, then use the [Matlab load_xdf](https://github.com/sccn/xdf) tools to read it in later.

I can't help you with BCILAB for online analysis but I know people use it. It seems like it has quite a few bugs from being neglected.

**Comment by hkn1304** [comment]
Yes, i would like to store real time data as a variable in Matlab workspace. So that i can process afterwards. when i try to run lsl_inlet, it asks me input which i am not aware of.  Is there a tutorial for this?

**Comment by dmedine** [comment]
A good place to start:

https://github.com/sccn/labstreaminglayer/tree/master/LSL/liblsl-Matlab/examples

**Comment by hkn1304** [comment]
Hello,

I am using Emotiv Epoc v2.0.0.20(older) for some reason and can't use Emotiv Epoc App(as this requires v.3.3.0)

That's why i am using Openvibe acquisition server to collect data from Epoc into LabRecorder and LSL and when i type vis_stream it seems to be working and i can see the data coming in the plot .

When everything is set up as explained above, i am running "ReceiveDataInChunks under "liblsl-Matlab/examples" without any modification(i don't if i need to change and type openvibeSignal which i see in the Lab recorder when i streamed data from openvibe into) it just says:
![image](https://cloud.githubusercontent.com/assets/22224352/26286678/0fc898ee-3e74-11e7-9e24-86ab0e6c7f33.png)
and nothing shows up.

However, i thought i found a way. I thought i need 2 Matlab; 1 for "SendingDataInChunks" and other for "ReceiveDataInChunks" and hit run simultaneously and it started to give something to me but i wasn't quite sure. Codes:
![image](https://cloud.githubusercontent.com/assets/22224352/26286718/ace0be4a-3e74-11e7-90af-d9e5621740ea.png)
![image](https://cloud.githubusercontent.com/assets/22224352/26286721/c03d5a70-3e74-11e7-905e-b27547632818.png)
I could get something in my CommandWindow but still can't store the older chunk values in a variable as it moves forward before i stopped. I got the impression from  randn(8,50) line in "SendingDataInChunks" i was getting virtual values that aren't related to Epoc.

I told you all the story. Now my questions are:
1. Is Emotiv App must in order to get data from Epoc ?(through openvibe and Labrecorder,even if i see Openvibesignal in Labrecorder)
2. Is there a way to get Epoc data under these conditions and what should i do in this case?
3. How can i store ex-values in a variable when the new chunk is available?

Any further help would be much appreciated. 
Many thanks in advance.

**Comment by dmedine** [comment]
SendDataInChunks and ReceiveDataInChunks are example programs that 
demonstrate how to use the matlab wrapper for liblsl. They don't 
actually /do/ anything other than setup a fake stream (in the case  of 
send) and receive values (in the case of receive).


On 21.05.17 21:36, hkn1304 wrote:
>
>  1. Is Emotiv App must in order to get data from Epoc ?(through
>     openvibe and Labrecorder,even if i see Openvibesignal in Labrecorder)
>
You can use either the Emotiv App or openvibe to stream the data via LSL
>
>  1. Is there a way to get Epoc data under these conditions and what
>     should i do in this case?
>
You can use LabRecorder to record the data. If you want to interact with 
the data in real time, you need to write software to do this (e.g. with 
python or matlab using LSL), or else create a BCI with openvibe, 
neuropype, etc.
>
>  1. How can i store ex-values in a variable when the new chunk is
>     available?
>
Line 19 in ReceiveDataInChunks:
[chunk, stampes] = inlet.pull_chunk();
